packages/dds/merge-tree/Obliterate.md

```diff
@@ -0,0 +1,276 @@
+
+Much of the same logic that goes into conflicting local + remote removal will need to be applied for obliterate.
+Nothing stands out as a conceptual issue or hurdle in this realm, though. Just tricky conditionals.
+
+Once the op is acked, the behavior in the [Remote perspective](#remote-perspective) section suffices for any further concurrent segments.
+
+### Other aspects
+
+TODO: Fill out this section in detail and with as much care as was taken above. For now (draft PR for early feedback), here is my current thinking on the areas I called out:
+
+- Any changes to direct fields of tree nodes themselves (either new data or changes to bookkeeping of existing data)
+> I think `{ obliteratedSeq?: number, localObliteratedSeq?: number }` suffices here. We'll of course need to make sure it's updated where appropriate (segment splits, checks for segment merges, etc.)
+- How the feature interacts with an increasing collab window and zamboni
+> I don't expect many conceptual issues here given the parallels with removedSeq, but need to do a code audit.
+- Impact on the partial lengths scheme
+> This does deserve its own section. Current partial lengths bookkeeping works for removal by adding various adjustment entries based on segments' `removedSeq` and overlapping removes fields. We need to update the bookkeeping on initial obliterate handling as well as on immediate-obliterate-on-inserting-walk cases.
+- Bookkeeping and handling of overlapping removals (note some may be obliterates and some may not be)
+> This was partially covered above. Not sure what else I want to add, but leaving it for now.
+- Reconnection
+> We might need to do fixup on the local state of the merge tree on reconnection by clearing rebased obliterate ops. I don't think the problem is intractible, and can always be done unperformantly in V1 since reconnection of an obliterate should be relatively uncommon.
+- Snapshotting impact
+> We'll need to ensure that obliterated info ends up on segments within the collab window. Otherwise I don't see issues here.
+
+## Endpoint Behavior
+
+One important consideration is what happens near the endpoints of the removed range.
+There are two general possibilities: either the obliterate expands to include segments inserted
+adjacent to the endpoint, or it does't.
+
+```typescript
+
+/** @sealed */
+interface IEndpointMergeStrategy {
+  // TODO
+}
+
+/** @sealed */
+interface IRangeMergeStrategy {
+  start: IEndpointMergeStrategy;
+  end: IEndpointMergeStrategy;
+}
+
+```
+
+## Public API
+
+The public API of sequence will need to be updated for users to leverage the obliterate operation.
+
+```typescript
+class SharedSegmentSequence<TInterval extends IInterval> {
+  public obliterateRange(start: number, end: number, merger: IRangeMergeStrategy = expand)
+}
+```
+
+## Relation to "Move"
+
+TODO: Move is incredibly tricky because unlike removal, it can nest to arbitrarily complex levels. Certain sequences of moves don't even have clear outcomes,
```

Comment:
i'm not sure it we will ever do move or not, but i think i'd like to keep open the possibility, and specifically, i'd like to reuse ops and props with obliterate, rather than have a distinct set for move and obliterate, basically obliterate is a subset of move we are supporting. it may complicate the discussion above, so i don't think we need to mention it everywhere, but when it comes to actual code and data on the wire it should make sense if we implement move later.

i haven't thought about naming very much, but maybe we don't call it move or obliterate. i like the succinctness of move, but people may get the wrong idea about what we support. probably need to spend some time with thesaurus to see if we can find a good phrasing.

Reply:
I agree. I had meant to call this out in the document (that op format should make sense for any sort of move operation we implement so we don't need to refactor again if/when we do decide to implement). Will include in a future draft.